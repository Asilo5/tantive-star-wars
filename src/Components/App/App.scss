*,
*:after,
*:before {
  box-sizing: border-box;
}

.App {
  // background-image: url('../../images/TantiveIVToydaria-SL.png');
  background-size: cover;
  background-repeat: no-repeat;
  height: 100vh;
  width: 100%;
  text-align: center;
}

.bb8-loading {
    height: 30em;
    margin-top: 14em;
    position: relative;
    right: 8em;
}
 
