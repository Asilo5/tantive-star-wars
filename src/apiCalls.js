export const getMovies = () => {
  return fetch('https://swapi.co/api/films/')
    .then(res => res.json())
    .then(movies => {
      return movies.results.map((movie) => {
        const firstTenCharacters = movie.characters.slice(0, 10)
        return {
          title: movie.title,
          episode_id: movie.episode_id,
          release_date: movie.release_date.slice(0, 4),
          characters: firstTenCharacters,
          opening_crawl: movie.opening_crawl
        }
      });
    });
  };