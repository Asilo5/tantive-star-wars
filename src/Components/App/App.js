import React, {Component} from 'react';
import { Route, Switch } from 'react-router-dom';
import Login from '../Login/Login';
import './App.scss';
import { getMovies, getCharacters } from '../../apiCalls';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import CharactersContainer from '../CharactersContainer/CharactersContainer';

class App extends Component {
  constructor() {
    super()
    this.state = {
      movies: [],
      characters: [],
      userName: '',
      userQuote: '',
      userRank: '',
      isLoading: true
    }
  }

  userInfo = ({name, quote, rank}) => {
     this.setState({
      userName: name,
      userQuote: quote,
      userRank: rank
     })
  }

  componentDidMount() {
    getMovies()
     .then(movies => movies.sort((a, b) => a.episode_id - b.episode_id))
     .then(movies => this.setState({ movies, isLoading: false}))
     .catch(err => console.log(err))
  }

  setCharacters = (charactersUrl) => {
    getCharacters(charactersUrl)
    .then(characterData => this.setState({characters: characterData}))
  }
  

  render() {
    const { isLoading, userName, userQuote, userRank } = this.state;
    return (
      <main className="App">
        <Switch>
          <Route exact path='/' render={() => <Login userInfo={this.userInfo}/> } />
          {isLoading ? <img className='bb8-loading' src='https://media.giphy.com/media/eEbiAqk9YUT5e/giphy.gif' alt='BB8 giff' /> : 
          (<Route exact path='/movies' render={() => <MoviesContainer movies={this.state.movies} userName={userName} userQuote={userQuote} userRank={userRank} setCharacters={this.setCharacters}/> } />)}
          <Route path='/characters' render={() => <CharactersContainer characters={this.state.characters} /> } />
        </Switch>
      </main>
    );
  }
}

export default App;
