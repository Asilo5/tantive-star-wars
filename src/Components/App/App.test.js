import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { getMovies, getCharacters } from '../../apiCalls'
jest.mock('../../apiCalls.js')

describe('App', () => {
  let wrapper,
      mockMovies;

  beforeEach(() => {
    getMovies.mockImplementation(() => {
      return Promise.resolve([{}, {}])
    });

    getCharacters.mockImplementation(() => {
      return Promise.resolve([{}, {}])
    });

    // mockMovies = [
    //     {
    //       title: 'The Phantom Menace', 
    //       episode_id: 1, 
    //       release_date: '1999',
    //       characters: ['person'], 
    //       opening_crawl: 'Turmoil has engulfed...'
    //     }, 
    //     {
    //       title: 'Attack of the Clones', 
    //       episode_id: 2, 
    //       release_date: '2002',
    //       characters: ['person'], 
    //       opening_crawl: 'Turmoil has engulfed...'
    //     }
    // ]
    
    // getMovies.mockImplementation(() => {
    //   return Promise.resolve(mockMovies);
    // });
    wrapper = shallow(<App />)
  });

  it('should match a snapshot of all the information being passed in correctly', () => {
      expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('should retrieve movies after mounting', () => {
      expect(getMovies).toHaveBeenCalled();
    });        
  });

  it('should have default values of empty strings for userName, userQuote and userRank', () => {
    expect(wrapper.state().userName).toEqual('');
    expect(wrapper.state().userQuote).toEqual('');
    expect(wrapper.state().userRank).toEqual('');
  });

  it('should have a default value of true for isLoading', () => {
    expect(wrapper.state().isLoading).toEqual(false);
  });

  it('should set the state of userName, userQuote, userRanking', () => {
    wrapper.setState({ userName: 'name', userQuote: 'quote', userRank: 'rank' });
  });

  // it.skip('should update state with movies when getMovies is called', async () => {
  //    await wrapper.instance().getMovies();
  //    expect(getMovies).toHaveBeenCalled();
  // });

  it('should retrieve the characters', async () => {
    await wrapper.instance().setCharacters();
    expect(getCharacters).toHaveBeenCalled();
  });

  describe('favourite testing', () => {
    it.skip('should toggle favourites based on an id', () => {
      wrapper.instance().toggleFavourites();
      expect(wrapper.state('favourites').favourites).toEqual()
    });

    it.skip('should be able to add a favourite to state based on an it', () => {
      wrapper.instance().addFavourites();
      expect(wrapper.state('favourites').favourites).toEqual('')
    });

    it.skip('should be able to delete a favourite to state based on an it', () => {
      wrapper.instance().deleteFavourites();
      expect(wrapper.state('favourites').favourites).toEqual('')
    })
  });


});
