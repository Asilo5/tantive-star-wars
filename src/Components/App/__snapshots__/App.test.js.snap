// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot of all the information being passed in correctly 1`] = `
<main
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies"
      render={[Function]}
    />
    <Route
      path="/characters"
      render={[Function]}
    />
  </Switch>
</main>
`;
